app/models/asset_group.rb

```diff
@@ -0,0 +1,24 @@
+class AssetGroup
+  def self.from_accounts(accounts)
+    total_nav = accounts.sum { |account| account.balance.cents }
+
+    accounts.group_by(&:accountable_type).map do |type, accounts|
+      accounts_total_value = accounts.sum { |account| account.balance.cents }
+
+      new(
+        Accountable.from_type(type),
+        accounts,
+        percentage_held: (accounts_total_value / total_nav.to_d * 100).round(2)
+      )
+    end
+  end
+
+  attr_reader :name, :type, :total_asset_value, :percentage_held, :param
+
+  def initialize(type, accounts, percentage_held: nil)
+    @type = type
+    @param = type.model_name.param_key.gsub("_", "-")
+    @total_asset_value = accounts.sum(&:balance)
+    @percentage_held = percentage_held
+  end
+end
```

Comment:
I feel like there is some overlap here between this and the accountable types themselves, and I wonder if some of this would be better suited to the Accountable concern. I'm also not sure if it leads to some confusion as we were talking about accounts, but now we're talking about assets?

Is there a distinct difference between an AssetGroup, and a given accountable type like Account::Credit?

Perhaps we could move total_value, and percentage_held into the concern as methods or derive some further concern that allows the retrieval of these values? Maybe AssetGroup is a concern?

I feel like an interface like the following might be preferable:
```
Account::Credit.total_value(user)
Account::Property.percentage_held(user)
```

The user/family argument could default to Current.user, so needn't be passed every time further cleaning the interface. It could also perhaps - at some point - accept a collection so we may be able to query for the total value of all accounts of a given type held in Maybe or for a segment of users/families.

I'd also consider adding a method on the accounts relation to group by type and return the real class for us since it feels like a common thing we need to do.

Could lead to something like this:
```
<% Current.user.accounts.by_type.each do |type| %>
  <%= type.total_value %> (<%= type.percentage_held %>)
<% end %>
```

Reply:
I get what you mean with the naming, think `AccountGroup` might be a better naming.
`AccountGroup` is a View model, kind of like a Presenter pattern.

The goal here is to group them, and store information about the group.
eg an AccountGroup model contains information of a particular type like `Account::Credit`

It allows us to extend the information to the next part of the UI.

![Screenshot 2024-02-10 at 11 54 41 PM](https://github.com/maybe-finance/maybe/assets/3957744/0e7a9f3c-e8a2-465e-927c-10b57444ff91)

Apologies if I don't understand your approach, as grouping by type and returning a type will not allow us to aggregate the data of a particular Account type like Account::Credit.

And this pattern would perform an N+1 query since they do not have context of the total account value of the total account value.
```
Account::Credit.total_value(user)
Account::Property.percentage_held(user)
```

Reply:
Yeah good points, I was mostly thinking about the interface rather than how it would be queried underneath. There might be a way to avoid the N+1 problem, but not sure it's worth the added complexity cost.

Thinking of it as a view model makes sense to me. Still not sure on the name though - one thing that came to mind was the concept of a Portfolio that takes all your Accounts, but I'll avoid bike shedding too much.

Reply:
Yeah the naming is hard to nail down. Initial thought with AssetGroup was because of the section title, "Total assets".